### PR TITLE
Add initial implementation.

### PR DESCRIPTION
**Pull request by lrhn**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.buildlog
```
Comment (kevmoo) [inline .gitignore]: Less is more here: https://dart.dev/guides/libraries/private-files

File: README.md [hunk 2]
```diff
@@ -1,4 +1,10 @@
 # characters
```
Comment (kevmoo) [inline README.md]: Remove superfluous title in readme. Redundant on both github page and pub site

File: travis.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+language: dart
```
Comment (kevmoo) [inline travis.yml]: Why this AND `.travis.yml`

File: .travis.yml [hunk 4]
```diff
@@ -1,4 +1,6 @@
 language: dart
 dart_task:
+- test: --platform vm
```
Comment (kevmoo) [inline .travis.yml]: Consider combining these? Booting up and running `pub get` costs a minute for each run.

File: .travis.yml [hunk 5]
```diff
@@ -1,4 +1,6 @@
...
+- test: --platform chrome
 - dartanalyzer
 - dartfmt
```
Comment (kevmoo) [inline .travis.yml:9]: Consider adding

```yaml
dart
- dev
- 2.4.0
```

Good to test on oldest supported SDK

File: .gitignore (hunk 1 above)
Comment (lrhn) [reply .gitignore]: I just copied this from some other repo. :)

I'll just go wtih
* dart-tool/
* .packages
* pubspec.lock

I'm not "build"-ing anything.

File: .travis.yml (hunk 4 above)
Comment (lrhn) [reply .travis.yml]: What would the syntax of that be? I can't see anything useful on https://docs.travis-ci.com/user/languages/dart/

File: .travis.yml (hunk 5 above)
Comment (lrhn) [reply .travis.yml:9]: Done

File: travis.yml (hunk 3 above)
Comment (lrhn) [reply travis.yml]: No reason. Removing.

File: README.md (hunk 2 above)
Comment (lrhn) [reply README.md]: Done.

**Comment by lrhn** [comment]
Should probably mention in readme that this is *experimental*.

File: .travis.yml (hunk 4 above)
Comment (kevmoo) [reply .travis.yml]: `--platform vm,chrome`

File: .travis.yml [hunk 6]
```diff
@@ -1,4 +1,9 @@
 language: dart
+dart:
+- dev
+- 2.4.0
 dart_task:
+- test: --platform vm
+- test: --platform chrome
 - dartanalyzer
 - dartfmt
```
Comment (kevmoo) [inline .travis.yml:9]: Also add

```yaml
# Only building master means that we don't run two builds for each pull request.
branches:
  only: [master]

cache:
 directories:
   - $HOME/.pub-cache
```

**Review by kevmoo — APPROVED** [review]

**Pull request merged**